tests/unit/ember-data-test.js

```diff
@@ -0,0 +1,37 @@
+import Ember from 'ember';
+import { test, moduleForModel } from 'ember-qunit';
+import stateFor from 'ember-state-services/state-for';
+
+let registry  = new Ember.Registry();
+let container = new Ember.Container(registry);
+let mockStatePOJO = { sample: 'example' };
+const registryOpts = { singleton: true, instantiate: false };
+
+moduleForModel('user', {
+  // needs: []
+  beforeEach() {
+    this.registry.register('state:test-state-pojo', mockStatePOJO, registryOpts);
+  }
+});
+
+test('that the key value can be an EmberData model', function(assert) {
+  var modelA = this.subject({
+    name: 'Bobby'
+  });
+  var modelB = this.subject({
+    name: 'Lisa'
+  });
+  let mockObject = Ember.Object.extend({
+    model: modelA,
```

Comment:
do you intend to put modelA on the prototype ? I suspect `model` and `container` should be provided via create.